subsys/power/pm_ctrl.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <kernel.h>
+#include <string.h>
+#include <soc.h>
+#include <device.h>
+#include "policy/pm_policy.h"
+
+#define LOG_LEVEL CONFIG_PM_LOG_LEVEL /* From power module Kconfig */
+#include <logging/log.h>
+LOG_MODULE_DECLARE(power);
+
+struct pm_ctrl_info {
+	int pm_state;
+	int count;
+};
+
+static struct pm_ctrl_info pm_ctrl[] = {
+	{SYS_PM_LOW_POWER_STATE, 0},
+	{SYS_PM_DEEP_SLEEP, 0},
+};
+
+void sys_pm_ctrl_lock_state(int state)
+{
+	if (state == SYS_PM_LOW_POWER_STATE) {
+		pm_ctrl[0].count++;
```

Comment:
What about using atomics here? In such case we will not need any locks.
Also, please add asserts catching overflow and underflow.

Reply:
Ok will use atomics. thanks.